zebra-rpc/src/methods/get_block_template_rpcs/get_block_template.rs

```diff
@@ -271,7 +271,6 @@ where
         unreachable!("unmatched response to a mempool::FullTransactions request")
     };
 
-    // Check that the mempool and state were in sync when we made the requests
```

Comment:
This deleted comment seems important, and it's unrelated to this PR

Reply:
It was very important and mistakenly deleted when reverting the changes in this file. Fixed in https://github.com/ZcashFoundation/zebra/pull/6510/commits/e6c6c0f33593095568ffef531379d445f2baa290, thank you for catching that.